tensorflow/lite/micro/python/interpreter/tests/test_model.py

```diff
@@ -0,0 +1,296 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Create a sample model for testing purposes.
+
+A simple quantized int8 model with single ADD operation.
+"""
+
+import flatbuffers
+import numpy as np
+
+from tflite_micro.tensorflow.lite.python import schema_py_generated as schema_fb
+
+TFLITE_SCHEMA_VERSION = 3
+
+
+def _generate_quantization_parameters(builder, rmin, rmax, qmin, qmax):
+  """Tensor-wise quantization parameter generation
+  """
+
+  def caclulate_scale(rmin, rmax, qmin, qmax):
+    return (rmax - rmin) / (qmax - qmin)
+
+  def calculate_zero_point(rmin, rmax, qmin, qmax):
+    """Zero point (zp) = int(qmin - rmin/s) or int(qmax - rmax/s);
+    use the one with smallest rounding error
+    """
+    scale = caclulate_scale(rmin, rmax, qmin, qmax)
+    zp_from_min = qmin - rmin / scale
+    zp_from_min_error = abs(qmin) + abs(rmin / scale)
+    zp_from_max = qmax - rmax / scale
+    zp_from_max_error = abs(qmax) + abs(rmax / scale)
+    zp = zp_from_min
+    if zp_from_max_error < zp_from_min_error:
+      zp = zp_from_max
+    if zp < qmin:
+      return qmin
+    if zp > qmax:
+      return qmax
+    return round(zp)
+
+  if rmin > 0:
+    raise ValueError("The quantization range [rmin, rmax] must include 0")
+    # Range info
+  schema_fb.QuantizationParametersStartMinVector(builder, 1)
+  builder.PrependFloat32(rmin)
+  quant_min_offset = builder.EndVector(1)
+  schema_fb.QuantizationParametersStartMaxVector(builder, 1)
+  builder.PrependFloat32(rmax)
+  quant_max_offset = builder.EndVector(1)
+  # scale
+  schema_fb.QuantizationParametersStartScaleVector(builder, 1)
+  scale = caclulate_scale(rmin, rmax, qmin, qmax)
+  builder.PrependFloat32(scale)
+  quant_scale_offset = builder.EndVector(1)
+  # zero point
+  schema_fb.QuantizationParametersStartZeroPointVector(builder, 1)
+  zp = calculate_zero_point(rmin, rmax, qmin, qmax)
+  builder.PrependInt64(zp)
+  quant_zero_point_offset = builder.EndVector(1)
+  # put into the buffer
+  schema_fb.QuantizationParametersStart(builder)
+  schema_fb.QuantizationParametersAddMin(builder, quant_min_offset)
+  schema_fb.QuantizationParametersAddMax(builder, quant_max_offset)
+  schema_fb.QuantizationParametersAddScale(builder, quant_scale_offset)
+  schema_fb.QuantizationParametersAddZeroPoint(builder,
+                                               quant_zero_point_offset)
+  quantization_offset = schema_fb.QuantizationParametersEnd(builder)
+  return quantization_offset
+
+
+def build_mock_flatbuffer_model():
```

Comment:
I'm sorry for the late review, this PR fell off my radar. 

A few comments around maintainability:
1. This is a hefty function to create a simple model with add. Can anything here be parameterized and abstracted away so that the next time we want to create another simple model with a different op, we don't have to replicate this or even attempt to understand the details here? For example, maybe we can break this long function into multiple helper functions?
2. I can see this utility being reused over and over for perhaps different or other use cases. Perhaps it's a good time for `tensorflow/lite/micro/python/utils`?
3. Can we have clearer function names and filenames? For example, if this file or this function's main export is a simple add model, we could make it super specific like `build_mock_add_flatbuffer_model()` so that it sets a good example for future additions.